modules/utils/include/hephaestus/utils/timing/stop_watch.h

```diff
@@ -27,13 +27,17 @@ class StopWatch {
   [[nodiscard]] auto stop() -> TargetDurationT;
 
   /// \return Currently running lap time:
-  /// - time elapsed from max[most recent start(), most recent lap()] to now.
+  /// - time elapsed from max[most recent start(), most recent lapse()] to now.
   /// - Cast to desired duration.
   /// - Doesn't stop the watch.
   template <typename TargetDurationT = StopWatch::DurationT>
   [[nodiscard]] auto lapse() -> TargetDurationT;
 
-  /// Stop and reset accumlated information.
+  /// \return Elapsed time since the last start() cast to the desired duration.
+  template <typename TargetDurationT = StopWatch::DurationT>
```

Comment:
I think the clock template parameter should be common to `StopWatch`, and all `StopWatch` member functions should not have a clock template parameter.

As is, you could define one clock for the StopWatch, and another one for `elapsed`, `stop`, ... This seems like a logic error.

Reply:
the clock is indeed the same this is the type of the returned duration you want. in this way you do not have to manually cast between milliseconds and microseconds